index.js

```diff
@@ -1,131 +1,146 @@
 var path = require('path'),
     fs = require('fs'),
-    parser = new require('xml2js').Parser()
+    parser = new require('xml2js').Parser();
 
 
-var commandArgs = {}
+var commandArgs = {};
 
 process.argv.forEach(function(arg) {
     if (arg.indexOf('--save') !== -1) {
-        commandArgs.save = arg.split('=')[1] || 'junit.html'
+        commandArgs.save = arg.split('=')[1] || 'junit.html';
     }
     if (arg.indexOf('--folder') !== -1) {
-        commandArgs.folder = arg.split('=')[1] || './test_data'
+        commandArgs.folder = arg.split('=')[1] || './test_data';
     }
     if (arg.indexOf('--port') !== -1) {
-        commandArgs.port = arg.split('=')[1] || 9090
+        commandArgs.port = arg.split('=')[1] || 9090;
     }
-})
+});
 
-var folder = commandArgs.folder || './test_data'
+var folder = commandArgs.folder || './test_data';
+folder = folder.replace('\\', '/');
 
 if (folder.charAt(0) == '.' && folder.charAt(1) == '/') {
-    folder = folder.replace('./', '')
+    folder = folder.replace('./', '');
 }
 
-console.log(folder)
-
-if (folder.charAt(folder.length - 1) !== '/') {
-    folder += '/'
-}
+var isFolder = fs.lstatSync(folder).isDirectory();
 
+console.log(folder);
 
+if (isFolder && folder.charAt(folder.length - 1) !== '/') {
+    folder += '/';
+}
 
 function endsWith(str, suffix) {
     return str.indexOf(suffix, str.length - suffix.length) !== -1;
 }
 
 function getFiles(folder) {
-    var files = []
-    fs.readdirSync(folder).forEach(function(file) {
-        if (fs.lstatSync(folder + '/' + file).isDirectory()) {
-            files = files.concat(getFiles(folder + '/' + file))
-        } else {
-            files.push(folder + '/' + file)
-        }
-    })
+    var files = [];
+    if (isFolder) {
+        fs.readdirSync(folder).forEach(function (file) {
+            if (fs.lstatSync(folder + '/' + file).isDirectory()) {
+                files = files.concat(getFiles(folder + '/' + file));
+            } else {
+                files.push(folder + '/' + file);
+            }
+        });
+    } else {
+        files.push(folder);
+    }
 
     return files.filter(function(file) {
-        var suffix = '.xml'
+        var suffix = '.xml';
         return file.indexOf(suffix, file.length - suffix.length) !== -1
-    })
+    });
 }
 
 function jsonResults() {
-    var files = getFiles(folder)
+    var files = getFiles(folder);
+    var finalResults = {};
 
-    var finalResults = {}
+    // Read all files in the folder (or that one file)
```

Comment:
if you feel a comment is needed then please wrap this in a function instead (similarlly see comment above)